### PR TITLE
Thread precision through np.convolve & np.correlate

### PR DESCRIPTION
**Pull request by jakevdp**
This should alleviate TPU flakes without having to set the tolerance so high.

### CODE REVIEW

File: tests/lax_numpy_test.py [hunk 1]
```diff
@@ -1155,8 +1155,8 @@ def testConvolutions(self, xshape, yshape, dtype, mode, rng_factory, jnp_op, onp
     rng = rng_factory()
     args_maker = lambda: [rng(xshape, dtype), rng(yshape, dtype)]
     onp_fun = partial(onp_op, mode=mode)
-    jnp_fun = partial(jnp_op, mode=mode)
-    tol = 1e-2 if jtu.device_under_test() != "tpu" else 0.5
+    jnp_fun = partial(jnp_op, mode=mode, precision=lax.Precision.HIGHEST)
```
Comment (froystig) [inline tests/lax_numpy_test.py:1158]: If I understand correctly, with this change, we will test `jax.numpy` only at `precision=HIGHEST`. Can we continue to exercise and test the other precision modes—especially the default—possibly at a different tolerance? A test parameter might be useful.

Comment (jakevdp) [reply tests/lax_numpy_test.py:1158]: I believe that the precision only affects TPU computations, and I don't think the default precision test with a tolerance of 50% is giving us much in terms of useful test coverage.

Comment (froystig) [reply tests/lax_numpy_test.py:1158]: Better understood now. Thanks!

Do we know that `precision` will never affect other devices in the future? Setting it only if `jtu.device_under_test() == "tpu"` might make sense unless we're certain about that.

Comment (jakevdp) [reply tests/lax_numpy_test.py]: Done

**Review by froystig — APPROVED** [review]

**Pull request merged**